resources/assets/js/videos/videoContainer.vue

```diff
@@ -551,7 +554,10 @@ export default {
         },
         fetchVideoContent(video) {
             let videoPromise = new Vue.Promise((resolve) => {
-                this.video.addEventListener('canplay', resolve);
+                this.video.addEventListener('canplay', () => {
+                    this.checkCORSProperty();
+                    resolve();
+                });
             });
```

Comment:
This would be more "promise-y":

```js
let videoPromise = new Vue.Promise((resolve) => {
    this.video.addEventListener('canplay', resolve);
});
videoPromise.then(this.checkCORSProperty);
```